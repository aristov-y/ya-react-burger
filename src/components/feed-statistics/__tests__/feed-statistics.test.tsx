import React from 'react';
import { create, act, ReactTestRenderer } from 'react-test-renderer';
import FeedStatistics from '../index';
import { useSelector } from 'react-redux';

const store = {
  feed: {
    total: 2002,
    totalToday: 34,
    feed: [{
      status: 'pending',
      createdAt: '2020-01-01T10:12:00',
      number: 2
    }, {
      status: 'done',
      createdAt: '2020-01-01T10:13:00',
      number: 3
    }, {
      status: 'pending',
      createdAt: '2020-01-01T10:14:00',
      number: 4
    }, {
      status: 'done',
      createdAt: '2020-01-01T10:10:00',
      number: 1
    }]
  }
}

beforeEach(() => {
  (useSelector as jest.Mock).mockImplementation(fn => fn(store));
});

describe('FeedStatisticOrders', function () {
  it('should render', function () {
    let component: ReactTestRenderer;
    act(() => {
      component = create(<FeedStatistics />)
    });

    expect(component.toJSON()).toMatchSnapshot();
  });
});
