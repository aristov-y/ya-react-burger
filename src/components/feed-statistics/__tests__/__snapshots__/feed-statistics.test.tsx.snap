// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FeedStatisticOrders should render 1`] = `
<div
  className="FeedStatistic text"
>
  <div
    className="FeedStatisticOrders"
  >
    <div
      className="Orders"
    >
      <span
        className="text text_type_main-medium"
      >
        Готовы:
      </span>
      <div
        className="OrdersList OrdersCompleted"
      >
        <span
          className="text text_type_digits-default"
        >
          000003
        </span>
        <span
          className="text text_type_digits-default"
        >
          000001
        </span>
      </div>
    </div>
    <div
      className="Orders"
    >
      <span
        className="text text_type_main-medium"
      >
        В работе:
      </span>
      <div
        className="OrdersList"
      >
        <span
          className="text text_type_digits-default"
        >
          000002
        </span>
        <span
          className="text text_type_digits-default"
        >
          000004
        </span>
      </div>
    </div>
  </div>
  <div
    className="FeedStatisticTotal"
  >
    <span
      className="text_type_main-medium"
    >
      Выполнено за все время:
    </span>
    <span
      className="text text_type_digits-large"
    >
      2002
    </span>
  </div>
  <div
    className="FeedStatisticTotal"
  >
    <span
      className="text_type_main-medium"
    >
      Выполнено за сегодня:
    </span>
    <span
      className="text text_type_digits-large"
    >
      34
    </span>
  </div>
</div>
`;
